diff -ru libsndfile-1.0.28.orig/configure.ac libsndfile-1.0.28/configure.ac
--- libsndfile-1.0.28.orig/configure.ac	2017-04-02 10:52:02.000000000 +0200
+++ libsndfile-1.0.28/configure.ac	2020-03-06 22:34:37.995899134 +0100
@@ -285,6 +285,7 @@
 HAVE_EXTERNAL_XIPH_LIBS=0
 EXTERNAL_XIPH_CFLAGS=""
 EXTERNAL_XIPH_LIBS=""
+EXTERNAL_XIPH_REQUIRE=""
 
 # Check for pkg-config outside the if statement.
 PKG_PROG_PKG_CONFIG
@@ -322,6 +323,11 @@
 
 		EXTERNAL_XIPH_CFLAGS="$FLAC_CFLAGS $OGG_CFLAGS $VORBIS_CFLAGS $VORBISENC_CFLAGS $SPEEX_CFLAGS"
 		EXTERNAL_XIPH_LIBS="$FLAC_LIBS $OGG_LIBS $VORBIS_LIBS $VORBISENC_LIBS $SPEEX_LIBS "
+		EXTERNAL_XIPH_REQUIRE="flac ogg vorbis vorbisenc"
+
+		if test x$ac_cv_speex = "xyes" ; then
+			EXTERNAL_XIPH_REQUIRE="$EXTERNAL_XIPH_REQUIRE speex"
+		fi
 	else
 		echo
 		AC_MSG_WARN([[*** One or more of the external libraries (ie libflac, libogg and]])
@@ -627,6 +633,7 @@
 
 AC_SUBST(EXTERNAL_XIPH_CFLAGS)
 AC_SUBST(EXTERNAL_XIPH_LIBS)
+AC_SUBST(EXTERNAL_XIPH_REQUIRE)
 AC_SUBST(SRC_BINDIR)
 AC_SUBST(TEST_BINDIR)
 
diff -ru libsndfile-1.0.28.orig/sndfile.pc.in libsndfile-1.0.28/sndfile.pc.in
--- libsndfile-1.0.28.orig/sndfile.pc.in	2016-09-10 10:08:27.000000000 +0200
+++ libsndfile-1.0.28/sndfile.pc.in	2020-03-06 22:34:29.345899091 +0100
@@ -5,8 +5,8 @@
 
 Name: sndfile
 Description: A library for reading and writing audio files
-Requires: 
+Requires:
+Requires.private: @EXTERNAL_XIPH_REQUIRE@
 Version: @VERSION@
 Libs: -L${libdir} -lsndfile
-Libs.private: @EXTERNAL_XIPH_LIBS@
-Cflags: -I${includedir} 
+Cflags: -I${includedir}
